### PR TITLE
initial

### PR DESCRIPTION
**Pull request by sanchezl**
Initial check-in before enabling CI & e2e. 
Built images lack the labels needed to be added to the installer payload.

### CODE REVIEW

File: manifests/0000_40_kube-storage-version-migrator-operator_00_crd.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: storageversionmigrations.migration.k8s.io
+  annotations:
+    "api-approved.kubernetes.io": "https://github.com/kubernetes/kubernetes/pull/0000"
```
Comment (sanchezl) [inline manifests/0000_40_kube-storage-version-migrator-operator_00_crd.yaml]: @deads2k this annotation is required. please advise on appropriate value.

File: pkg/operator/targetcontroller/controller.go [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+)
+
+const (
+	workQueueKey = "key"
+)
+
+type TargetController struct {
```
Comment (sttts) [inline pkg/operator/targetcontroller/controller.go]: 3 liner what this does

File: pkg/operator/targetcontroller/controller.go [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+	forceRequeue, err := c.syncKubeStorageVersionManager(spec, status)
+	if forceRequeue && err != nil {
+		c.queue.AddRateLimited(workQueueKey)
+	}
+
+	return nil
+}
+
+func NewTargetController(
```
Comment (sttts) [inline pkg/operator/targetcontroller/controller.go]: usually constructors are above, below the type

File: pkg/operator/targetcontroller/sync.go [hunk 4]
```diff
@@ -0,0 +1,192 @@
+package targetcontroller
+
+import (
+	"fmt"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	"strings"
```
Comment (sttts) [inline pkg/operator/targetcontroller/sync.go]: order

File: vendor/github.com/NYTimes/gziphandler/.gitignore [hunk 5]
```diff
@@ -0,0 +1 @@
+*.swp
```
Comment (sttts) [inline vendor/github.com/NYTimes/gziphandler/.gitignore]: please don't mix bumps and actual code into one commit. Makes it hard to review.

File: manifests/0000_40_kube-storage-version-migrator-operator_00_crd.yaml (hunk 1 above)
Comment (sanchezl) [reply manifests/0000_40_kube-storage-version-migrator-operator_00_crd.yaml]: I've set it to `"unapproved, unsupported"`.

File: pkg/operator/starter.go [hunk 6]
```diff
@@ -0,0 +1,96 @@
+package operator
+
+import (
+	"fmt"
+	"github.com/openshift/cluster-kube-storage-version-migrator-operator/pkg/operator/targetcontroller"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/kubernetes"
+	"os"
+	"time"
```
[the remark this answers is not in the document]
Comment (sanchezl) [reply pkg/operator/starter.go]: Done

File: pkg/operator/targetcontroller/controller.go (hunk 2 above)
Comment (sanchezl) [reply pkg/operator/targetcontroller/controller.go]: Done

File: pkg/operator/targetcontroller/controller.go (hunk 3 above)
Comment (sanchezl) [reply pkg/operator/targetcontroller/controller.go]: Done

File: pkg/operator/targetcontroller/sync.go (hunk 4 above)
Comment (sanchezl) [reply pkg/operator/targetcontroller/sync.go]: Done

File: vendor/github.com/NYTimes/gziphandler/.gitignore (hunk 5 above)
Comment (sanchezl) [reply vendor/github.com/NYTimes/gziphandler/.gitignore]: Done

**Pull request merged**